### PR TITLE
chore: 241 finish wiring data service api endpoint to hmi server proxy

### PR DESCRIPTION
**Pull request by dgauldie**
# Description

Added all proxies, models, and resources to link with data-service

Resolves #(issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide reproduction instructions and list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/PersonResource.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		return proxy.createPerson(person);
+	}
+
+	@GET
+	@Path("/{id}")
+	public Response getPerson(
+		@PathParam("id") final String id
+	) {
+		return proxy.getPerson(id);
+	}
```
Comment (YohannParis) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/PersonResource.java:82]: I don't see an integration with _Keyclock_ PostGres Table.
Is that something we will do in another PR?

Comment (dgauldie) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/PersonResource.java:82]: this is data-service wiring only

Comment (YohannParis) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/PersonResource.java:82]: I thought the Proxy was the wiring
```
hmi-client -> hmi-server (/api/* endpoints) -> hmi-server (proxy to DS) -> data-service
```

Comment (dgauldie) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/PersonResource.java]: yeah, I'm only wiring up endpoints that come for the data service. Currently, the data service has "Person" endpoints. I'm sure this will change later on, and as far as I know, there is no mock data for this endpoint anyway.

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ConceptResource.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+@Tag(name = "Concept REST Endpoints")
+public class ConceptResource {
+
+	@Inject
+	@RestClient
+	ConceptProxy proxy;
+
+	@GET
+	public Response searchConcept(
+		@QueryParam("term") final String term,
```
Comment (dvince2) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ConceptResource.java:26]: `@QueryParam` parameters are optional fields will pass in `null` if not defined.  Do we have guarantees that `proxy.searchConcept` can accept `null` values?  If not we may want to set these using `@DefaultValue`

Comment (dvince2) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ConceptResource.java:26]: This question holds for all of the XYZResource.java classes in the dataservice directory -  I won't spam on each of them.

**Review by dvince2 — DISMISSED** [review]
Unsure if we need to use default values or if the underlaying methods will be handling null queryparams?

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/PersonResource.java (hunk 1 above)
Comment (fivegrant) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/PersonResource.java]: @YohannParis the Person endpoint is for managing info about people related to a resource that aren't necessarily users (e.g. I uploaded this Dataset but the maintainer is actually some scientist in the UK). 

The script is populating the endpoint with one entry: 
```
        {
            "name": "Adam Smith",
            "email": "Adam@test.io",
            "org": "Uncharted",
            "website": "",
            "is_registered": True,
        }
```

**Review by bigglesandginger — APPROVED** [review]
LGTM

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ConceptResource.java (hunk 2 above)
Comment (dgauldie) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ConceptResource.java:26]: the default values for ints and strings should be valid for these query params

**Review by mwdchang — APPROVED** [review]
